compiler/rustc_resolve/src/late/diagnostics.rs

```diff
@@ -2181,6 +2181,11 @@ impl<'a: 'ast, 'ast> LateResolutionVisitor<'a, '_, 'ast> {
         generics_span: Span,
         params: &[ast::GenericParam],
     ) {
+        if generics_span.is_empty() {
```

Comment:
Hm... maybe we should put this check into the `deletion_span` closure instead, and just return `None`? It's still kind of nice to be able to emit this lint.

Reply:
How about we delay the check to this position, so we can emit more possible lint

https://github.com/rust-lang/rust/blob/5348a89a77a49f6f57c13c0edf7f493fd410159a/compiler/rustc_lint/src/context.rs#L842

Reply:
That works too probably